### PR TITLE
Reset pswd

### PR DESCRIPTION
**Pull request by Riwan25**

### CODE REVIEW

File: apps/users-api/src/routes/forgotPassword.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    const emailData = Email$.parse({
+        to: email,
+        subject: 'Verify your email',
+        html: `<a href="${url.toString()}">Click here to reset your password</a>`
+    });
+    sendEmail(emailData);
+}
+
+const Body$ = z.object({
+    email: z.string().email(),
```
Comment (Melv1C) [inline apps/users-api/src/routes/forgotPassword.ts:25]: Email$

File: apps/users-api/src/routes/forgotPassword.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                include: {
+                    preferences: true
+                }
+            });
+            if (!user) {
+                res.status(404).json({ message: 'No account found with this email'});
+                return;
+            }
+            sendRestPasswordEmail(user.email, generateResetPasswordToken(TokenData$.parse(user)));
+            res.send('Email sent');
```
Comment (Melv1C) [inline apps/users-api/src/routes/forgotPassword.ts:47]: wait the email to realy be sent

File: apps/users-api/src/routes/resetPassword.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    token: EncodeToken$,
+});
+
+const Body$ = z.object({
+    newPassword: z.string(),
+});
+
+router.post(
+    '/reset-password',
+    parseRequest(Key.Query, Query$),
```
Comment (Melv1C) [inline apps/users-api/src/routes/resetPassword.ts:20]: parse body

File: apps/users-api/src/routes/resetPassword.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    async (req, res) => {
+        try {
+            const { token } = Query$.parse(req.query);
+            const tokenData = verifyResetPasswordToken(token);
+            const { newPassword } = Body$.parse(req.body);
+            if (!tokenData) {
```
Comment (Melv1C) [inline apps/users-api/src/routes/resetPassword.ts:26]: order

File: apps/users-api/src/routes/resetPassword.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+                res.status(401).json({ message: 'Invalid or expired token' });
+                return;
+            }
+            const user = await prisma.user.findUnique({
+                where: {
+                    email: tokenData.email
```
Comment (Melv1C) [inline apps/users-api/src/routes/resetPassword.ts:32]: id

File: apps/users-api/src/routes/resetPassword.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+                }
+            });
+            if (!user) {
+                res.status(404).json({ message: 'No account found with this email' });
```
Comment (Melv1C) [inline apps/users-api/src/routes/resetPassword.ts:36]: User not found

File: apps/users-api/src/routes/resetPassword.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+                return;
+            }
+            const hashedPassword = await bcrypt.hash(newPassword, 10);
```
Comment (Melv1C) [inline apps/users-api/src/routes/resetPassword.ts:39]: Extract the function

File: apps/users-api/src/routes/resetPassword.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    async (req, res) => {
+        try {
+            const { token } = Query$.parse(req.query);
+            const tokenData = verifyResetPasswordToken(token);
+            const { newPassword } = Body$.parse(req.body);
+            if (!tokenData) {
+                res.status(401).json({ message: 'Invalid or expired token' });
+                return;
+            }
+            const user = await prisma.user.findUnique({
```
Comment (Melv1C) [inline apps/users-api/src/routes/resetPassword.ts:30]: Pas utile

File: shared-packages/utils/src/jwtToken.ts [hunk 9]
```diff
@@ -14,6 +14,10 @@ export const generateVerificationToken = (tokenData: TokenData) => {
     return EncodeToken$.parse(jwt.sign(tokenData, process.env.VERIFY_EMAIL_TOKEN_SECRET!, { expiresIn: '7d' }));
 }
 
+export const generateResetPasswordToken = (tokenData: TokenData) => {
+    return EncodeToken$.parse(jwt.sign(tokenData, process.env.RESET_PASSWORD_TOKEN_SECRET!, { expiresIn: '1h' }));
```
Comment (Melv1C) [inline shared-packages/utils/src/jwtToken.ts:18]: 15min

File: apps/users-api/src/routes/forgotPassword.ts (hunk 1 above)
Comment (Riwan25) [reply apps/users-api/src/routes/forgotPassword.ts:25]: bha non ça existait pas

**Review by Melv1C — APPROVED** [review]

**Pull request merged**